code/datums/components/udder.dm

```diff
@@ -60,17 +61,21 @@
  */
 /obj/item/udder
 	name = "udder"
+	///typepath of reagent produced by the udder
+	var/reagent_produced_typepath = /datum/reagent/consumable/milk
 	///how much the udder holds
 	var/size = 50
 	///mob that has the udder component
 	var/mob/living/udder_mob
 	///optional proc to callback to when the udder generates milk
 	var/datum/callback/on_generate_callback
 
-/obj/item/udder/Initialize(mapload, udder_mob, on_generate_callback)
+/obj/item/udder/Initialize(mapload, udder_mob, on_generate_callback, custom_milk_reagent)
 	src.udder_mob = udder_mob
 	src.on_generate_callback = on_generate_callback
 	create_reagents(size, REAGENT_HOLDER_ALIVE)
+	if(custom_milk_reagent)
+		reagent_produced_typepath = custom_milk_reagent
```

Comment:
if you set the default on the component, you won't need this if statement. I also think you should change this to src.reagent_produced_typepath = reagent_produced_typepath